packages/vuetify/src/components/VDatePicker/util/determineWeeknumber.ts

```diff
@@ -0,0 +1,58 @@
+const hourInMilliseconds = 60 * 1000
+const dayInMilliseconds = 24 * 60 * hourInMilliseconds
+
+function createUTCDate (year: number, month = 0, day = 1) {
+  var date
+  if (year < 100 && year >= 0) {
+    date = new Date(Date.UTC(year, month, day))
+    if (isFinite(date.getUTCFullYear())) {
+      date.setUTCFullYear(year)
+    }
+  } else {
+    date = new Date(Date.UTC(year, month, day))
+  }
+
+  return date
+}
+
+function firstWeekOffset (year: number, firstDayOfWeek: number, firstDayOfYear: number) {
+  const fwd = 7 + firstDayOfWeek - firstDayOfYear
+  const fwdlw = (7 + createUTCDate(year, 0, fwd).getUTCDay() - firstDayOfWeek) % 7
+
+  return -fwdlw + fwd - 1
+}
+
+function dayOfYear (determineDate: Date, newYear: Date) {
```

Comment:
`newYear` can be calculated from `date` (`newYear = createUTCDate(date.getFullYear())`)

Also does this code works fine with changing timezones? I think it's safer to use the previous method https://github.com/vuetifyjs/vuetify/pull/10926/files#diff-fad712b1d24c3d982c0bd796f2ee28d1L60 (and provide year/month/day as arguments instead of Date object)

Reply:
Refactored dayOfYear method